File: Swift-C++-IR-Comparison/test/enumerations/Swift/SimpleEnum.swift
enum CompassPoint {
    case north
    case south
    case east
    case west
}
@main
struct MyMain {

    static func main() {

        let first = wayfinder(direction: CompassPoint.east)
        let second = wayfinder(direction: CompassPoint.north)

    }

    static func wayfinder(direction: CompassPoint) -> Int {
        var x = 0
        switch direction {
        case .north:
            x += 1
        case .south:
            x += 2
        case .east:
            x += 3
        case .west:
            x += 412
        }
        return x
    }
}

File: AutoTranspiledTests/Swift-to-Cpp/try-1/SimpleEnum.cpp

enum CompassPoint
{
    north,
    south,
    east,
    west
};
 
int wayfinder(CompassPoint direction)
{
    int x = 0;
    switch (direction)
    {
    case north:
        x += 1;
        break;
    case south:
        x += 2;
        break;
    case east:
        x += 3;
        break;
    case west:
        x += 412;
        break;
    }
    return x;
}
 
int main()
{
    int first = wayfinder(east);
    int second = wayfinder(north);
 
    return 0;
}